### ISSUE TITLE
Properly handle nested objects

### ISSUE THREAD

**Issue by aaronpk** [indieweb/microformats2-ruby#31]
It seems the parser is not handling nested objects properly.

For example, this URL: http://aaronparecki.com/notes/2014/07/04/2/indiewebcamp-latergram

It appears the comment authors and comment URLs all show up under the main h-entry when in reality they should be under children of the main h-entry as their own h-cite objects.

Compare the [result of the PHP parser](http://pin13.net/mf2/?url=http%3A%2F%2Faaronparecki.com%2Fnotes%2F2014%2F07%2F04%2F2%2Findiewebcamp-latergram)

**Comment by jeena** [comment]
I tried to parse http://tantek.com/ and it crashed the parser on something like this:
```html
<div class="h-entry">
  <p class="u-comment h-cite">
    test .
  </p>
</div>
```
with:

    URI::InvalidURIError: bad URI(is not URI?): test .

It would be nice if the parser at least wouldn't crash.

**Comment by veganstraightedge** [comment]
@jeena It seems like the parser is pretty much abandoned by the G5 folks. We got paid to build it (when I was still there) as a part of a larger product. But if it does as much as G5 needs and they're otherwise too busy, it's not likely to get the attention it deserves.

If you're able and willing to submit a pull request with a patch, I could apply it for you.

Unfortunately, (like too many open source projects) this doesn't really have a maintainer anymore. :/

**Comment by jeena** [comment]
Oh ok, that's sad, but understandable. Perhaps one could write that somewhere into the README so people know and perhaps someone will be able to take it over. I'm not sure I will be able to fix something like this but if then I will make a pull request.

**Comment by veganstraightedge** [comment]
@jeena @aaronpk  I believe this is fixed in `3.0`. I just did a test and it looks good to me. Please upgrade and run your comparison too. Re-open this issue if necessary.
